### PR TITLE
[SELS-Feature][BE] Setup Middleware (Auth/Error)

### PR DESCRIPTION
**Pull request by IanUrriza10**
## [Asana Link]
https://app.asana.com/0/1203212597546540/1203223794310934
## [Expected Output]
New files in middleware 
authMiddleware and errorMiddleware

Will throw a `404 Request not found on all routes except get('/')

## [Notes]
Added custom middleware for route protection and custom error handling
Will be used on Basic Routing (BE)
## [Screenshots]
image

### CODE REVIEW

File: backend/src/middleware/authMiddleware.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      const token = req.headers.authorization.split(" ")[1];
+
+      const decoded = jwt.verify(token, process.env.JWT_SECRET);
+
+      //Get user from the token
+      // req.user = await //find decoded.id minus the password
+
+      next();
+    } catch (error) {
+      console.log(error);
```
Comment (roseaugusto) [inline backend/src/middleware/authMiddleware.js:20]: please remove console.logs

File: backend/src/middleware/authMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    req.headers.authorization &&
+    req.headers.authorization.startsWith("Bearer")
+  ) {
+    try {
+      //get token from header
+      const token = req.headers.authorization.split(" ")[1];
+
+      const decoded = jwt.verify(token, process.env.JWT_SECRET);
+
+      //Get user from the token
```
Comment (roseaugusto) [inline backend/src/middleware/authMiddleware.js:15]: please remove this

File: backend/src/middleware/authMiddleware.js (hunk 1 above)
Comment (IanUrriza10) [reply backend/src/middleware/authMiddleware.js]: Fixed

File: backend/src/middleware/authMiddleware.js (hunk 2 above)
Comment (IanUrriza10) [reply backend/src/middleware/authMiddleware.js]: Fixed

File: backend/src/server.js [hunk 3]
```diff
@@ -27,5 +28,11 @@ app.post("/", (req, res) => {
   res.status(200).json({ msg: "Backend Post Working" });
 });
 
+app.get("/1", (req, res) => {
```
Comment (roseaugusto) [inline backend/src/server.js:31]: what is this route for? if only sample, maybe put something beneficial in the future like 404 page

Comment (IanUrriza10) [reply backend/src/server.js:31]: For testing the Error middleware

Comment (roseaugusto) [reply backend/src/server.js:31]: @IanUrriza10 my suggestion would be to put a comment so it's understandable to be removed in the future or maybe change this route to 404 page to be used later as well

Comment (IanUrriza10) [reply backend/src/server.js]: Noted po. I have converted it to catch all types of requests and send a 404 not found if needed.

**Review by roseaugusto — APPROVED** [review]

**Pull request merged**